crypto/src/main/java/pm/gnosis/crypto/utils/Extensions.kt

```diff
@@ -0,0 +1,24 @@
+package pm.gnosis.crypto.utils
+
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asEthereumAddressString
+import pm.gnosis.utils.removeHexPrefix
+import pm.gnosis.utils.toHexString
+
+fun Solidity.Address.asEthereumAddressChecksumString(): String {
+    val address = asEthereumAddressString().removeHexPrefix()
+    val keccak = Sha3Utils.keccak(address.toByteArray()).toHexString()
```

Comment:
I was looking into a solution that avoid the `toHexString` to save some memory allocation ... but the solution is not really nicer :P 

```
fun Solidity.Address.asEthereumAddressChecksumString(): String =
    asEthereumAddressString().removeHexPrefix().run {
        val checksum = Sha3Utils.keccak(toByteArray())
        foldIndexed(StringBuilder("0x"), { j, stringBuilder, c ->
            stringBuilder.append(
                when {
                    c in '0'..'9' -> c
                    checksum.hexCharValue(j) >= 8 -> c.toUpperCase()
                    else -> c.toLowerCase()
                }
            )
        }).toString()
    }

private fun ByteArray.hexCharValue(position: Int) =
    (get(position / 2).toInt() ushr (4 * ((position + 1) % 2))) and 0x0F
```